Rename some sizevars methods for clarity

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #105714
* __->__ #105585

The guard functions require you to ALREADY KNOW that a particular
condition holds.  If you don't know (you want to guard on an expression
being a particular value, and then get access to that value), use
the evaluate functions.

I renamed the functions that don't abide by this:

```
guard_min -> evaluate_min
guard_max (deleted, no uses)
guard_static_shape -> evaluate_static_shape
guard_static_shapes -> evaluate_static_shapes
```

Some added comments.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

ezyang committed:
Rename some sizevars methods for clarity

The guard functions require you to ALREADY KNOW that a particular
condition holds.  If you don't know (you want to guard on an expression
being a particular value, and then get access to that value), use
the evaluate functions.

I renamed the functions that don't abide by this:

```
guard_min -> evaluate_min
guard_max (deleted, no uses)
guard_static_shape -> evaluate_static_shape
guard_static_shapes -> evaluate_static_shapes
```

Some added comments.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

[ghstack-poisoned]

voznesenskym commented on torch/_inductor/sizevars.py:
```diff
@@ -317,7 +334,11 @@ def guard_lt(self, left: Expr, right: Expr) -> None:
+    # The evaluate functions evaluate some symbolic sympy expression
+    # (NB: not necessarily an Expr) and return what the concrete result
+    # is, guarding on the expression being that result
+
```
nit newline

ezyang replied:
Intentional, there is more than one evaluate function, deleting the newline associates it with the wrong function

voznesenskym commented on torch/_inductor/sizevars.py:
```diff
@@ -307,6 +311,19 @@ def statically_known_multiple_of(self, numerator: Expr, denominator: Expr) -> bo
+    # condition holds.  If you don't know (you want to guard on an expression
+    # being a particular value, and then get access to that value), use
+    # the evaluate functions.
+
```
nit newline? also maybe we should just always go through speculation? requiring that they already know can be an assert at the callsite...?

ezyang committed:
Update on "Rename some sizevars methods for clarity"


The guard functions require you to ALREADY KNOW that a particular
condition holds.  If you don't know (you want to guard on an expression
being a particular value, and then get access to that value), use
the evaluate functions.

I renamed the functions that don't abide by this:

```
guard_min -> evaluate_min
guard_max (deleted, no uses)
guard_static_shape -> evaluate_static_shape
guard_static_shapes -> evaluate_static_shapes
```

Some added comments.

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

ezyang commented:
@pytorchbot merge

ezyang commented:
@pytorchbot merge -i
